Add Land Transaction Tax, Land and Buildings Transaction Tax and business rates

nikhilwoodruff opened the pull request:
Adds logic and parameters for Land Transaction Tax and LBTT (Wales and Scotland's Stamp Duty equivalents, respectively).

MaxGhenis commented on openfisca_uk/parameters/revenue_scotland/lbtt/non_residential.yaml:
```diff
@@ -0,0 +1,25 @@
+description: Rates on residential property puchases
```
```suggestion
description: Rates on Scotland's non-residential property purchases
```

MaxGhenis commented on openfisca_uk/parameters/revenue_scotland/lbtt/residential/rate.yaml:
```diff
@@ -0,0 +1,34 @@
+description: Rates on residential property puchases
```
```suggestion
description: Rates on Scotland's residential property purchases
```

MaxGhenis commented on openfisca_uk/tests/policy/baseline/wra/land_transaction_tax/land_transaction_tax.yaml:
```diff
@@ -0,0 +1,44 @@
+    country: WALES
+  output:
+    land_transaction_tax: 127_750
+
```
extra line

MaxGhenis commented on openfisca_uk/variables/finance/income.py:
```diff
@@ -205,8 +205,10 @@ class hbai_household_net_income(Variable):
         tax = household("household_tax", period)
-        ignored_taxes = household("expected_stamp_duty", period) + household(
-            "corporate_tax_incidence", period
+        ignored_taxes = (
```
can this use `add`?

MaxGhenis commented on openfisca_uk/variables/finance/tax/tax.py:
```diff
@@ -27,6 +27,7 @@ def formula(household, period):
         personal_taxes = household.sum(household.members("tax", period))
         household_taxes = (
             household("expected_stamp_duty", period)
+            + household("expected_land_transaction_tax", period)
```
same - `add`

MaxGhenis commented on openfisca_uk/variables/revenue_scotland/lbtt.py:
```diff
@@ -0,0 +1,113 @@
+from openfisca_uk.model_api import *
+
+
+class lbtt_on_residential_property(Variable):
```
so i prefer lowercase acronyms but we should probably be consistent for now and switch them all over if we decide to do that

MaxGhenis commented on openfisca_uk/variables/revenue_scotland/lbtt.py:
```diff
@@ -0,0 +1,113 @@
+    value_type = float
+    unit = "currency-GBP"
+
+    def formula(household, period, parameters):
```
is there a way to reduce code duplication between this and sdlt?

MaxGhenis commented on openfisca_uk/variables/revenue_scotland/lbtt.py:
```diff
@@ -0,0 +1,113 @@
+        non_residential_purchase_tax = lbtt.non_residential.calc(price)
+        # Tax on non-residential rents
+        cumulative_rent = household("cumulative_non_residential_rent", period)
+        rent = household("non_residential_rent", period)
```
what would a household's non-residential rent include? is this in the microdata?

MaxGhenis commented on openfisca_uk/variables/revenue_scotland/lbtt.py:
```diff
@@ -0,0 +1,113 @@
+
+class land_and_buildings_transaction_tax(Variable):
+    label = "Land and Buildings Transaction Tax"
+    documentation = "Total tax liability for LBTT"
```
```suggestion
    documentation = "Total tax liability for Scotland's LBTT"
```

MaxGhenis commented on openfisca_uk/variables/revenue_scotland/lbtt.py:
```diff
@@ -0,0 +1,113 @@
+    unit = "currency-GBP"
+
+    def formula(household, period):
+        return household("lbtt_on_residential_property", period) + household(
```
`add`

MaxGhenis commented on openfisca_uk/variables/wealth/corporate.py:
```diff
@@ -21,14 +39,7 @@ class corporate_tax_incidence(Variable):
-        ).sum()
-        percentage_change = where(
-            total_wealth > 0, total_change / total_wealth, 0
+        total_change = household("corporate_stamp_duty", period) - household(
```
```suggestion
        return household("corporate_stamp_duty", period) - household(
```

MaxGhenis commented on openfisca_uk/variables/wealth/land.py:
```diff
@@ -11,22 +11,36 @@ class owned_land(Variable):
+    unit = "currency-GBP"
+
+    def formula(household, period):
+        return household("household_land_value", period) + household(
```
`add`

MaxGhenis commented on openfisca_uk/variables/wra/land_transaction_tax.py:
```diff
@@ -0,0 +1,94 @@
+class ltt_liable(Variable):
+    label = "Liable for Land Transaction Tax"
+    documentation = (
+        "Whether the household is liable to pay the Land Transaction Tax"
```
```suggestion
        "Whether the household is liable to pay the Wales Land Transaction Tax"
```

MaxGhenis commented on openfisca_uk/variables/wra/land_transaction_tax.py:
```diff
@@ -0,0 +1,94 @@
+
+
+class ltt_on_residential_property(Variable):
+    label = "LTT on residential property"
```
This PR is inconsistent on acronyms in labels vs documentation. I'd prefer keeping acronyms out of labels and putting them inside parens in the documentation. I'd also suggest putting the country in the documentation.

nikhilwoodruff commented:
@MaxGhenis I've just added the business rates variables/parameters and added all property taxes to the documentation. Here's the new page: 
[Property Taxes — OpenFisca-UK documentation.pdf](https://github.com/PolicyEngine/openfisca-uk/files/7745113/Property.Taxes.OpenFisca-UK.documentation.pdf)
